app/controlplane/pkg/data/group.go

```diff
@@ -369,20 +369,68 @@ func (g GroupRepo) Update(ctx context.Context, orgID uuid.UUID, groupID uuid.UUI
 }
 
 // SoftDelete soft-deletes a group by setting the DeletedAt field to the current time.
+// It also marks all group memberships as deleted and removes any pending invitations related to the group.
 func (g GroupRepo) SoftDelete(ctx context.Context, orgID uuid.UUID, groupID uuid.UUID) error {
-	// Softly delete the group by setting the DeletedAt field
-	_, err := g.data.DB.Group.UpdateOneID(groupID).
-		SetDeletedAt(time.Now()).
-		Where(group.OrganizationIDEQ(orgID), group.DeletedAtIsNil()).
-		Save(ctx)
-	if err != nil {
-		if ent.IsNotFound(err) {
-			return biz.NewErrNotFound("group")
+	return WithTx(ctx, g.data.DB, func(tx *ent.Tx) error {
+		now := time.Now()
+
+		// Softly delete the group by setting the DeletedAt field
+		_, err := tx.Group.UpdateOneID(groupID).
+			SetDeletedAt(now).
+			Where(group.OrganizationIDEQ(orgID), group.DeletedAtIsNil()).
+			Save(ctx)
+		if err != nil {
+			if ent.IsNotFound(err) {
+				return biz.NewErrNotFound("group")
+			}
+			return fmt.Errorf("failed to mark group as deleted: %w", err)
 		}
-		return err
-	}
 
-	return nil
+		// Mark as deleted all group memberships for this group
+		_, err = tx.GroupMembership.Update().
+			Where(
+				groupmembership.GroupID(groupID),
+				groupmembership.DeletedAtIsNil(),
+			).
+			SetDeletedAt(now).
+			Save(ctx)
+		if err != nil && !ent.IsNotFound(err) {
+			return fmt.Errorf("failed to mark group memberships as deleted: %w", err)
+		}
+
+		// Delete all memberships where this group is either the member or the resource
+		_, err = tx.Membership.Delete().Where(
+			membership.HasOrganizationWith(organization.ID(orgID)),
+			membership.Or(
+				membership.MemberID(groupID),
+				membership.And(
+					membership.ResourceID(groupID),
+					membership.ResourceTypeEQ(authz.ResourceTypeGroup),
```

Comment:
is this possible? a membership with type = group? Group memberships have their own table, right?

Reply:
replying myself, the maintainer role is stored here, thanks!

Reply:
Yes the maintainer one